cost/aws/old_snapshots/aws_delete_old_snapshots.pt

```diff
@@ -0,0 +1,263 @@
+    "headers": {
+      "User-Agent": "RS Policies",
+      "Content-Type": "text/xml"
+    },
+    "query_params": {
+      'Action': 'DescribeSnapshots',
+      'Version': '2016-11-15',
+      'Filter.1.Name': 'status',
+      'Filter.1.Value.1': 'completed',
+      'Filter.2.Name': 'owner-id',
+      'Filter.2.Value.1': ds_get_caller_identity[0]['account']
+    }
+  }
+EOS
+end
+
+script "js_filter_old_snapshots", type: "javascript" do
+  result "results"
+  parameters "snapshot_age", "ds_describe_snapshots", "param_exclude_tags"
+  code <<-EOS
+    console.log("Snapshot Age "+snapshot_age);
+    var param_exclude_tags_lower=[];
+    for(var j=0;j<param_exclude_tags.length;j++){
+        param_exclude_tags_lower[j] = param_exclude_tags[j].toString().toLowerCase();
+    }
+    var results = [];
+    for(var i=0; i < ds_describe_snapshots.length; i++){
+      var snapshot = ds_describe_snapshots[i];
+      // Check, if the tag present in entered param_exclude_tags, ignore the snapshot if tag matches/present
+      var tags = snapshot['tags'];
+      var isTagMatched = false;
+      var tagKeyValue = "";
+      for(var k=0; k < tags.length; k++){
+        tag = tags[k];
+        if((param_exclude_tags_lower.indexOf((tag['tagKey']).toLowerCase()) !== -1) || (param_exclude_tags_lower.indexOf((tag['tagKey']+'='+tag['tagValue']).toLowerCase()) !== -1)){
+          isTagMatched = true;  
+		}
+
+        // Constructing tags with comma separated to display in detail_template
+        if((tag['tagValue']).length > 0){
+          tagKeyValue = tagKeyValue+" , "+tag['tagKey']+'='+tag['tagValue'];  
+		}else{
+          tagKeyValue = tagKeyValue+" , "+tag['tagKey'];  
+		}
+	  }
+
+      if(tagKeyValue === "" || tagKeyValue === " " || tagKeyValue == ""){
+        tagKeyValue = "   < No Value >";
+      }
+
+      if(!(isTagMatched)){
+        var snapshotTime = Date.parse(snapshot['startTime']);
+        var daysOld = (new Date().getTime() - new Date(snapshotTime).getTime()) / (1000 * 3600 * 24);
+        if(daysOld > snapshot_age){
+          results.push({
+            snapshotId : snapshot['snapshotId'],
+            region : snapshot['region'],
+            tagKeyValue : (tagKeyValue.slice(2)),
+            daysOld : daysOld.toFixed(2)
+		  })
+		}
+	  }
+	}
+    results = _.sortBy(results,'region');
+    results = _.sortBy(results,'daysOld');
+  EOS
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "pol_utilization" do
+  validate $ds_filter_old_snapshots do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} rows containing AWS old snapshot data"
+    detail_template <<-EOS
+# AWS Old Snapshot
+| Region | Days Old | Snapshot Id | Tags |
+| ------ | ---------- | ------ | ------- |
+{{ range data -}}
+| {{.region}} | {{.daysOld}} | {{.snapshotId}} | {{ .tagKeyValue }} |  
```

Comment:
The days Old value has a decimal, it should be a whole number.

Reply:
Done !